### PR TITLE
Ensure consistency of stats server settings by moving the logic to global helm value

### PR DESCRIPTION
**Pull request by GrahamGoudeau**
Ensure that all stats server settings are consistent across open source and enterprise Gloo by moving the enabled logic into a global helper template. This change introduces a global value that all deployments will use as their default stats server settings: `global.glooStats.enabled`. This value can be overridden by an individual deployment by setting a field on that deployment. For example, to disable stats on discovery, you can do `discovery.deployment.stats.enabled=false`. Note that this change has converted the `stats` field on our deployment objects from a boolean to an object with an `enabled` boolean field inside of it.

### Testing
Imported the gloo chart into enterprise gloo and was able to similarly control the stats server enabled status using the helper template from open source gloo.
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1884

### CODE REVIEW

**Comment by ashleywang1** [comment]
I'm not sure why the Helm2 test is failing / is it possible that the test is not actually waiting for check-code-and-docs-gen to finish? Is this a flake?
/kick

**Comment by kdorosh** [comment]
This was fixed by https://github.com/solo-io/gloo/pull/2011 , which was picked up in https://github.com/solo-io/gloo/pull/2010/commits/289d2e5a48e7c7fb9de499614a6d718cd8f116d0

File: install/helm/gloo/templates/1-gloo-deployment.yaml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {{- $image := .Values.gloo.deployment.image }}
+{{- $statsContext := dict "Deployment" .Values.gloo.deployment "GlobalContext" .Values.global }}
+{{- $statsEnabled := (include "gloo.statsServerEnabled" $statsContext) -}}
```
Comment (kdorosh) [inline install/helm/gloo/templates/1-gloo-deployment.yaml]: prefer `merge` function

File: install/helm/gloo/templates/1-gloo-deployment.yaml [hunk 2]
```diff
@@ -19,7 +22,7 @@ spec:
     metadata:
       labels:
         gloo: gloo
-      {{- if .Values.gloo.deployment.stats }}
+      {{- if eq "true" $statsEnabled }}
```
Comment (kdorosh) [inline install/helm/gloo/templates/1-gloo-deployment.yaml]: should be able to just do `{{- if $stats.enabled }}`

File: install/test/helm_test.go [hunk 3]
```diff
@@ -167,6 +170,86 @@ var _ = Describe("Helm Test", func() {
...
+						Expect(ok).To(BeTrue(), fmt.Sprintf("Deployment %+v should be able to cast to a structured deployment", deployment))
+
+						if structuredDeployment.GetName() == "gloo" {
+							Expect(structuredDeployment.Spec.Template.ObjectMeta.Annotations).To(BeEmpty(), fmt.Sprintf("No annotations should be present on deployment %+v", structuredDeployment))
+						} else if structuredDeployment.GetName() == "discovery" {
+							for annotation, value := range normalPromAnnotations {
+								Expect(structuredDeployment.Spec.Template.ObjectMeta.Annotations[annotation]).To(Equal(value), fmt.Sprintf("Annotation %s should be set to %s on deployment %+v", deployment, annotation, value))
+							}
+						} else {
+							assert.Fail(GinkgoT(), fmt.Sprintf("Unexpected deployment found: %+v", structuredDeployment))
```
Comment (kdorosh) [inline install/test/helm_test.go]: gomega has a `Fail` function we can use here instead of pulling in a new dependency for an assert

**Pull request merged**